File: app/channels/lounge_channel.rb
class LoungeChannel < ApplicationCable::Channel
  def subscribed
    stream_from "lounge_channel_#{params['room']}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def start(data)
    Room.where(room_name: data['roomname']).update(rand_first_character: data['rand_first_character'])
    Room.where(room_name: data['roomname']).update(rand_last_character: data['rand_last_character'])
    User.where(roomname: data['roomname']).each do |player|
      player.update(randnum: rand(64))
    end
    x = 0
    User.where(roomname: data['roomname']).order(:randnum).each do |bigtetsu|
      bigtetsu.update(randorder: x)
      x += 1
    end



    ActionCable.server.broadcast "lounge_channel_#{params['room']}", roomname: data['roomname'], game: data['game'], rand_first_character: data['rand_first_character']
  end

  # def deleteroom(data)
  #   # if Room.created_time + 10 < Time.now
  #     ActionCable.server.broadcast "lounge_channel_#{params['room']}", roomname: data['roomname']
  #   # else

  #   # end
  # end

  # def leave(data)
  #   User.find(data['userid']).update(roomname: nil)
  #   ActionCable.server.broadcast 'lounge_channel', roomname: data['roomname'], userid: data['userid'], leaving: data['leaving']
  # end
end

File: app/javascript/channels/lounge_channel.js
import consumer from "./consumer"

const appLounge = consumer.subscriptions.create({channel: 'LoungeChannel', room: $('#room_info').data('room_name') }, {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log("Game Start!");
    if(data['game'] === 0 && $('#room_info').data('room_name') === data['roomname']){
      window.location.pathname = "rooms/" + data['roomname']
    }
  },

  start: function(roomname, game, rand_first_character, rand_last_character) {
    return this.perform('start', {
      roomname: roomname,
      game: game,
      rand_first_character: rand_first_character,
      rand_last_character: rand_last_character
    });
  },

  // deleteroom: function(roomname) {
  //   return this.perform('deleteroom', {roomname: roomname});
  // }
});

window.addEventListener('load', function () {
  let characters = "あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわがぎぐげござじずぜぞだづでどばびぶべぼぱぴぷぺぽきいち"
  let initial_rand = Math.floor( Math.random() * (70) );
  let final_rand = Math.floor( Math.random() * (70) );
  $('#start-btn').on('click', function() {
    appLounge.start($('#room_info').data('room_name'), 0, characters.substr(initial_rand, 1), characters.substr(final_rand, 1))
  });
});
